source/AMDSClientUi.cpp

```diff
@@ -0,0 +1,332 @@
+
+
+//	amptekIndex = new QSpinBox();
+
+
+	/// ==== layout the components ====
+	QGridLayout *mainLayout = new QGridLayout;
+	mainLayout->addWidget(hostLabel_, 0, 0);
+	mainLayout->addWidget(hostLineEdit_, 0, 1);
+	mainLayout->addWidget(portLabel_, 1, 0);
+	mainLayout->addWidget(portLineEdit_, 1, 1);
+	mainLayout->addWidget(new QLabel("Active Servers"), 2, 0);
+	mainLayout->addWidget(activeServerComboBox_, 2, 1);
+	mainLayout->addWidget(statusLabel_, 3, 0, 1, 2);
+	mainLayout->addWidget(buttonBox_, 4, 0, 1, 2);
+
+	QFormLayout *formLayout = new QFormLayout();
+	formLayout->addRow("Request Type", requestTypeComboBox_);
+	formLayout->addRow("Buffer", bufferNameListView_);
+	formLayout->addRow("Time1", time1Edit_);
+	formLayout->addRow("Time2", time2Edit_);
+	formLayout->addRow("Count1", count1Edit_);
+	formLayout->addRow("Count2", count2Edit_);
+	formLayout->addRow("Include Status", includeStatusDataCheckbox_);
+	formLayout->addRow("Enable Flattening", enableFlattenDataCheckbox_);
+	mainLayout->addLayout(formLayout, 5, 0, 1, 2);
+	formLayout->addRow("Active continuous connection", activeContinuousConnectionComboBox_);
+	mainLayout->addWidget(resultsEdit_, 6, 0, 1, 2);
+	setLayout(mainLayout);
+
+	setWindowTitle(tr("AMDS Client Example"));
+	portLineEdit_->setFocus();
+
+	/// ==== initialize the app controller ==============
+	clientAppController_ = new AMDSClientAppController();
+	connect(clientAppController_, SIGNAL(networkSessionOpening()), this, SLOT(onNetworkSessionOpening()));
+	connect(clientAppController_, SIGNAL(networkSessionOpened()), this, SLOT(onNetworkSessionOpened()));
+	connect(clientAppController_, SIGNAL(newServerConnected(QString)), this, SLOT(onNewServerConnected(QString)));
+	connect(clientAppController_, SIGNAL(serverError(int,QString,QString)), this, SLOT(onServerError(int,QString,QString)));
+	connect(clientAppController_, SIGNAL(requestDataReady(AMDSClientRequest*)), this, SLOT(onRequestDataReady(AMDSClientRequest*)));
+	connect(clientAppController_, SIGNAL(socketError(int, QString, QString)), this, SLOT(onSocketError(int, QString, QString)));
+
+	clientAppController_->openNetworkSession();
+}
+
+AMDSClientUi::~AMDSClientUi()
+{
+	disconnect(clientAppController_, SIGNAL(networkSessionOpening()), this, SLOT(onNetworkSessionOpening()));
+	disconnect(clientAppController_, SIGNAL(networkSessionOpened()), this, SLOT(onNetworkSessionOpened()));
+	disconnect(clientAppController_, SIGNAL(newServerConnected(QString)), this, SLOT(onNewServerConnected(QString)));
+	disconnect(clientAppController_, SIGNAL(serverError(int,QString,QString)), this, SLOT(onServerError(int,QString,QString)));
+	disconnect(clientAppController_, SIGNAL(requestDataReady(AMDSClientRequest*)), this, SLOT(onRequestDataReady(AMDSClientRequest*)));
+	disconnect(clientAppController_, SIGNAL(socketError(QString, QString)), this, SLOT(onSocketError(QString, QString)));
```

Comment:
Missing an int argument in this SIGNAL (should be socketError(int, QString, QString)).